Kokkos Ecosystem: 4.3.00

lucbv opened the pull request:
Update Kokkos ecosystem for 4.3.00 release

lucbv committed:
Kokkos Kernels: adding missing TPLs and pre-conditions

Adding variants and dependencies for rocBLAS and rocSPARSE.
Also adding a "when=" close to the TPL variants that prevents
enabling the TPLs in versions of the library when it was not
yet available.

lucbv committed:
Kokkos Kernels: remove comment for better format

lucbv committed:
Kokkos Kernels: adding cusolver and rocsolver for at version 4.3.00

lucbv committed:
Kokkos Ecosystem: updating packages for release 4.3.00

lucbv committed:
Kokkos: adding arch for SG2042

nmm0 commented on var/spack/repos/builtin/packages/kokkos/package.py:
```diff
@@ -119,6 +120,7 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
         "skylake": "BDW",
         "icelake": "SKX",
         "skylake_avx512": "SKX",
+        "sophgo2042": "RISCV_SG2042",
```
I think we should add this as an official spack architecture

lucbv replied:
I pushed a commit to remove this, it show appear soon. Once that's the case I think this will be ready to go.

cedricchevalier19 replied:
Perhaps we should also try to use archspec in Kokkos.

tldahlgren reviewed (dismissed):
Confirmed the two version sha256s.  Is it okay to merge this PR as is or should I wait for another maintainer to comment?

nmm0 commented:
> Confirmed the two version sha256s. Is it okay to merge this PR as is or should I wait for another maintainer to comment?

@tldahlgren I think we will probably split the added architecture (`sophgo2042`) into a separate PR since we probably want to add it to spack's list of known architectures

lucbv committed:
Removing sg2042 from spack_micro_arch_map

Removing it here and will work to add it in the proper generic spack location, likely: `spack/lib/spack/external/archspec/json/cpu/microarchitectures.json` ?

nmm0 reviewed (approved):
LGTM :)

tldahlgren commented:
> > Confirmed the two version sha256s. Is it okay to merge this PR as is or should I wait for another maintainer to comment?
> 
> @tldahlgren I think we will probably split the added architecture (`sophgo2042`) into a separate PR since we probably want to add it to spack's list of known architectures

That definitely would be better than adding it to this PR.
